File: src/coding_problems/grammarly/coding_golf/T.clj
(import java.time.LocalDate)

(defn l [i, d]
      (if (< i 43)
        (do
          (println (str "JDK "
                        (if (< i 5)
                          (str "1." i)
                          i)
                        (if (.isAfter (LocalDate/now) d)
                          " was"
                          " will be"
                          )
                        " released in "
                        (.format d
                                 (java.time.format.DateTimeFormatter/ofPattern "MMMM yyyy"))
                        ))
          (l
            (+ i 1)
            (.plusMonths d
                         (nth [0, 13, 22, 17, 21, 31, 27, 55, 32, 42] (+ i 1) 6))
            )
          )
        )
      )

(l 0 (LocalDate/of 1996, 1, 1))

File: src/coding_problems/grammarly/coding_golf/T.java
package coding_problems.grammarly.coding_golf;

import java.time.LocalDate;

interface T {
    static void main(String... a) {
        int[] s = {0, 13, 22, 17, 21, 31, 27, 55, 32, 42};
        var d = LocalDate.of(1996, 1, 1);
        for (int i = 0; i < 43; i++) {
            System.out.printf("JDK %s %s released in %s\n",
                              i < 5 ? "1." + i : "" + i,
                              LocalDate.now().isAfter(d = d.plusMonths(i < s.length ? s[i] : 6)) ? "was" : "will be",
                              d.format(java.time.format.DateTimeFormatter.ofPattern("MMMM yyyy")));
        }
    }
}